server/src/main/java/com/cloud/network/lb/LoadBalancingRulesManagerImpl.java

```diff
@@ -1598,65 +1602,73 @@ public LoadBalancer createPublicLoadBalancerRule(String xId, String name, String
         // LoadBalancer result = _elbMgr.handleCreateLoadBalancerRule(lb,
         // lbOwner, lb.getNetworkId());
         LoadBalancer result = null;
-        if (result == null) {
-            IpAddress systemIp = null;
-            NetworkOffering off = _entityMgr.findById(NetworkOffering.class, network.getNetworkOfferingId());
-            if (off.isElasticLb() && ipVO == null && network.getVpcId() == null) {
-                systemIp = _ipAddrMgr.assignSystemIp(networkId, lbOwner, true, false);
-                if (systemIp != null) {
-                    ipVO = _ipAddressDao.findById(systemIp.getId());
+        IpAddress systemIp = null;
+        NetworkOffering off = _entityMgr.findById(NetworkOffering.class, network.getNetworkOfferingId());
+
+        if (srcPortStart == 53 && ipVO.isSourceNat()) {
```

Comment:
can you factor 53 out as DNS_PORT?

Reply:
Sure @DaanHoogland

Reply:
For some reason checkstyle doesn't like that name, it's complaining:
Name 'DNS_PORT' must match pattern '^(s_)?[a-z][a-zA-Z0-9]*$'. [StaticVariableName]

I  think I am going to have to go with dnsPort on this one.

Reply:
Done